src/lib/homestore.cpp

```diff
@@ -145,6 +145,16 @@ bool HomeStore::start(const hs_input_params& input, hs_before_services_starting_
 }
 
 void HomeStore::format_and_start(std::map< uint32_t, hs_format_params >&& format_opts) {
+    auto total_pct_sum = 0.0f;
+    for (const auto& [svc_type, fparams] : format_opts) {
+        total_pct_sum += fparams.size_pct;
+    }
+
+    if (total_pct_sum > 100.0f) {
```

Comment:
https://github.com/eBay/HomeStore/blob/202f07fa57361f657d74a60c13c907c6391e256f/src/lib/device/device_manager.cpp#L222

since we use `round_up`, so there may be a case that every v_dev will get little more size of expected, so the final sum of all dev will cap the total size, right? should we use `round_down`